qiime/core/type/primitive.py

```diff
@@ -349,6 +349,24 @@ def encode(self, value):
 Float = _Float('Float')
 
 
+class _Bool(_Primitive):
+    _valid_predicates = {Arguments,}
+
+    def _is_element_(self, value):
+        return isinstance(value, bool)
+
+    def decode(self, string):
+        if string not in ('false', 'true'):
+            raise TypeError("%s is neither 'true' or 'false'")
+
+        return string == 'true'
+
+    def encode(self, value):
+        return str(value)
```

Comment:
Okay, thanks. Not sure how far to read into things yet :) but totally understand that much of this is still annealing. I'll go with @jairideout's suggestion right now. Okay with avoiding the `in` check for now given that `encode` may go away?